### ISSUE TITLE
RX 5500 not reporting temps, fan speed, (utilization) core clock/mem clock

### ISSUE THREAD

**Issue by goodspeed** [exelban/stats#451]
Can anyone else confirm this? 

The MSI RX 5500 XT is showing 0 values for temp, fan speed & GPU utilization. 

It worked for our other setup RX 580.

Edit: I'm using a Ryzen 5 2600 / Gigabte Aorus B450 I / MSI RX 5500 XT 8GB / AMD Power Gadget / Hackintosh Opencore 0.6.2

Edit 2: Pasted the text output

**Comment by exelban** [comment]
Hi. Could you please post the output of `ioreg -r -d 1 -w 0 -c "IOAccelerator"`?
It will help to debug if there is not this data in driver or something goes wrong in the Stats.

**Comment by goodspeed** [comment]
Sorry I lost access to the machine. Here's the output--

ioreg -r -d 1 -w 0 -c "IOAccelerator"

+-o AMDRadeonX6000_AMDNavi14GraphicsAccelerator  <class AMDRadeonX6000_AMDNavi14GraphicsAccelerator, id 0x100000461, registered, matched, active, busy 0 (317 ms), retain 118>
    {
      "AccelCaps" = 15
      "IODVDBundleName" = "AMDRadeonVADriver2"
      "IOGVAHEVCEncode" = "1"
      "ATY,intrev" = 21
      "AccelNativeDMARowByteAlignment" = 256
      "InternalStatisticsAccm" = {}
      "IOGeneralInterest" = "IOCommand is not serializable"
      "GPUDCCDisplayable" = Yes
      "InternalStatistics" = {}
      "MetalPluginClassName" = "AMDMTLGFX10Device"
      "GpuDebugPolicy" = 0
      "AGCInfo" = {"fBusyCount"=0,"poweredOffByAGC"=No,"fSubmissionsSinceLastCheck"=150996,"fLastSubmissionPID"=188}
      "dpm" = 1
      "MetalPluginName" = "AMDRadeonX6000MTLDriver"
      "PerformanceStatisticsAccum" = {"iosurfaceTextureCreationBytes"=0,"HWChannel SDMA0_PAGE | Commands Completed"=0,"HWChannel VCN0Dec | Commands Completed"=0,"Device Utilization %"=0,"orphanedNonReusableSysMemoryCount"=0,"swapCompleteVideoWaitTime"=0,"orphanedReusableSysMemoryHitRate"=0,"oolTextureCreationCount"=0,"surfaceCount"=18,"HWChannel KIQ | Commands Submitted"=0,"HWChannel SDMA1_PAGE | Commands Completed"=0,"context2DCount"=4,"surfaceCopyOutWaitTime"=0,"surfaceReadLockIdleWaitTime"=0,"stdTexturePageInBytes"=0,"swapBytesPerSample"=0,"finishCLWaitTime"=0,"gartCacheBytes"=33554432,"agpTextureCreationCount"=0,"stdTextureCreationBytes"=0,"textureCount"=2577,"HWChannel VCN0Enc | Commands Submitted"=0,"surfaceTextureCreationCount"=0,"agprefTextureCreationBytes"=0,"finish2DWaitTime"=0,"contextVideoCount"=0,"orphanedNonReusableSysMemoryBytes"=0,"surfaceBufferPageOutBytes"=0,"swapComplete2DWaitTime"=0,"contextGLCount"=2,"surfaceBufferTextureCreationCount"=0,"orphanedReusableSysMemoryBytes"=0,"orphanedNonReusableVidMemoryCount"=0,"HWChannel GFX | Commands Completed"=0,"clientSharedAllocatedBytes"=1290240,"HWChannel Compute1 | Commands Completed"=0,"clientGLWaitTime"=0,"freeSurfaceSwapBufferWaitTime"=0,"Fan Speed(%)"=0,"surfaceBufferReadOutBytes"=0,"volatileSurfaceCount"=0,"hardwareSubmitWaitTime"=0,"HWChannel SDMA1 | Commands Submitted"=0,"HWChannel Compute1 | Commands Submitted"=0,"inUseVidMemoryBytes"=435888128,"vramFreeBytes"=7461748736,"ioSurfacePageInBytes"=0,"HWChannel SDMA0 | Commands Submitted"=0,"finishAll2DWaitTime"=0,"recoveryCount"=0,"orphanedReusableVidMemoryCount"=576,"Core Clock(MHz)"=0,"orphanedNonReusableVidMemoryBytes"=0,"HWChannel SDMA0_PAGE | Commands Submitted"=0,"Memory Clock(MHz)"=0,"vramEvictionWaitTime"=0,"stdTextureCreationCount"=0,"gartSizeBytes"=6442450944,"GPU Activity(%)"=0,"HWChannel VCN0EncLLQ | Commands Submitted"=0,"oolTexturePageInBytes"=0,"texturePageOutBytes"=0,"freeToAllocGPUAddressWaitTime"=0,"gartMapInBytesPerSample"=0,"Fan Speed(RPM)"=0,"HWChannel VCN0Dec | Commands Submitted"=0,"HWChannel VCN0Enc | Commands Completed"=0,"oolTextureCreationBytes"=0,"finishGLWaitTime"=0,"HWChannel KIQ | Commands Completed"=0,"HWChannel SDMA1 | Commands Completed"=0,"freeDataBufferWaitTime"=0,"HWChannel SDMA1_PAGE | Commands Submitted"=0,"orphanedReusableVidMemoryHitRate"=96,"HWChannel SDMA0 | Commands Completed"=0,"inUseSysMemoryBytes"=132866048,"hardwareWaitTime"=0,"bufferSwapCount"=0,"swapCompleteGLWaitTime"=0,"agprefTextureCreationCount"=0,"agpTextureCreationBytes"=0,"dataBytesPerSample"=0,"HWChannel GFX | Commands Submitted"=0,"HWChannel Compute0 | Commands Completed"=0,"iosurfaceTextureCreationCount"=0,"bufferFlipCount"=0,"gartMapOutBytesPerSample"=0,"Temperature(C)"=0,"contextCLCount"=0,"surfaceBufferPageInBytes"=0,"freeSurfaceBackingWaitTime"=0,"HWChannel Compute0 | Commands Submitted"=0,"textureVolunteerUnloadBytes"=0,"ioSurfaceReadOutBytes"=0,"finishVideoWaitTime"=0,"HWChannel Compute2 | Commands Completed"=0,"gartUsedBytes"=132866048,"Total Power(W)"=0,"textureReadOutBytes"=0,"orphanedReusableSysMemoryCount"=0,"gartFreeBytes"=6309584896,"dataBufferCount"=0,"ioSurfacePageOutBytes"=0,"orphanedReusableVidMemoryBytes"=675520512,"surfaceSetShapeIdleWaitTime"=0,"HWChannel Compute2 | Commands Submitted"=0,"surfaceWriteLockIdleWaitTime"=0,"surfaceCopyInWaitTime"=0,"HWChannel VCN0EncLLQ | Commands Completed"=0}
      "PerformanceStatistics" = {"iosurfaceTextureCreationBytes"=90112,"HWChannel SDMA0_PAGE | Commands Completed"=1059,"HWChannel VCN0Dec | Commands Completed"=0,"Device Utilization %"=7,"orphanedNonReusableSysMemoryCount"=0,"swapCompleteVideoWaitTime"=0,"orphanedReusableSysMemoryHitRate"=0,"oolTextureCreationCount"=55,"surfaceCount"=18,"HWChannel KIQ | Commands Submitted"=0,"HWChannel SDMA1_PAGE | Commands Completed"=148,"context2DCount"=4,"surfaceCopyOutWaitTime"=0,"surfaceReadLockIdleWaitTime"=0,"stdTexturePageInBytes"=87244800,"swapBytesPerSample"=0,"finishCLWaitTime"=0,"gartCacheBytes"=33554432,"agpTextureCreationCount"=11,"stdTextureCreationBytes"=113197056,"textureCount"=2577,"HWChannel VCN0Enc | Commands Submitted"=0,"surfaceTextureCreationCount"=0,"agprefTextureCreationBytes"=0,"finish2DWaitTime"=0,"contextVideoCount"=0,"orphanedNonReusableSysMemoryBytes"=0,"surfaceBufferPageOutBytes"=0,"swapComplete2DWaitTime"=0,"contextGLCount"=2,"surfaceBufferTextureCreationCount"=0,"orphanedReusableSysMemoryBytes"=0,"orphanedNonReusableVidMemoryCount"=0,"HWChannel GFX | Commands Completed"=713,"clientSharedAllocatedBytes"=1290240,"HWChannel Compute1 | Commands Completed"=3,"clientGLWaitTime"=0,"freeSurfaceSwapBufferWaitTime"=0,"Fan Speed(%)"=0,"surfaceBufferReadOutBytes"=0,"volatileSurfaceCount"=0,"hardwareSubmitWaitTime"=0,"HWChannel SDMA1 | Commands Submitted"=13,"HWChannel Compute1 | Commands Submitted"=3,"inUseVidMemoryBytes"=435888128,"vramFreeBytes"=7461748736,"ioSurfacePageInBytes"=24576,"HWChannel SDMA0 | Commands Submitted"=8,"finishAll2DWaitTime"=0,"recoveryCount"=0,"orphanedReusableVidMemoryCount"=576,"Core Clock(MHz)"=0,"orphanedNonReusableVidMemoryBytes"=0,"HWChannel SDMA0_PAGE | Commands Submitted"=1059,"Memory Clock(MHz)"=0,"vramEvictionWaitTime"=0,"stdTextureCreationCount"=314,"gartSizeBytes"=6442450944,"GPU Activity(%)"=0,"HWChannel VCN0EncLLQ | Commands Submitted"=0,"oolTexturePageInBytes"=208814080,"texturePageOutBytes"=0,"freeToAllocGPUAddressWaitTime"=0,"gartMapInBytesPerSample"=0,"Fan Speed(RPM)"=0,"HWChannel VCN0Dec | Commands Submitted"=0,"HWChannel VCN0Enc | Commands Completed"=0,"oolTextureCreationBytes"=22232576,"finishGLWaitTime"=0,"HWChannel KIQ | Commands Completed"=0,"HWChannel SDMA1 | Commands Completed"=13,"freeDataBufferWaitTime"=0,"HWChannel SDMA1_PAGE | Commands Submitted"=148,"orphanedReusableVidMemoryHitRate"=96,"HWChannel SDMA0 | Commands Completed"=8,"inUseSysMemoryBytes"=132866048,"hardwareWaitTime"=0,"bufferSwapCount"=0,"swapCompleteGLWaitTime"=0,"agprefTextureCreationCount"=0,"agpTextureCreationBytes"=45056,"dataBytesPerSample"=0,"HWChannel GFX | Commands Submitted"=716,"HWChannel Compute0 | Commands Completed"=0,"iosurfaceTextureCreationCount"=28,"bufferFlipCount"=0,"gartMapOutBytesPerSample"=0,"Temperature(C)"=0,"contextCLCount"=0,"surfaceBufferPageInBytes"=0,"freeSurfaceBackingWaitTime"=0,"HWChannel Compute0 | Commands Submitted"=0,"textureVolunteerUnloadBytes"=0,"ioSurfaceReadOutBytes"=0,"finishVideoWaitTime"=0,"HWChannel Compute2 | Commands Completed"=8,"gartUsedBytes"=132866048,"Total Power(W)"=0,"textureReadOutBytes"=49152,"orphanedReusableSysMemoryCount"=0,"gartFreeBytes"=6309584896,"dataBufferCount"=0,"ioSurfacePageOutBytes"=0,"orphanedReusableVidMemoryBytes"=675520512,"surfaceSetShapeIdleWaitTime"=0,"HWChannel Compute2 | Commands Submitted"=8,"surfaceWriteLockIdleWaitTime"=0,"surfaceCopyInWaitTime"=0,"HWChannel VCN0EncLLQ | Commands Completed"=0}
      "IOGLBundleName" = "AMDRadeonX6000GLDriver"
      "IOPropertyMatch" = {"LoadAccelerator"=Yes}
      "IOKitDebug" = 0
      "IOSourceVersion" = "0.0.0.0.0"
      "IOGVAHEVCDecodeCapabilities" = {"VTPerProfileDetails"={"1"={"VTMaxDecodeLevel"=153},"2"={"VTMaxDecodeLevel"=153},"3"={"VTMaxDecodeLevel"=153}},"VTSupportedProfileArray"=(1,2,3)}
      "GPUDebugNullClientMask" = 0
      "IOGVAH264EncodeCapabilities" = {"VTQualityRating"=50,"VTRating"=350}
      "ATY,cbits" = 262144
      "IOOCDBundleName" = "AMDRadeonX6000GLDriver"
      "cail_properties" = {}
      "CFBundleIdentifier" = "com.apple.kext.AMDRadeonX6000"
      "IOAccelRevision" = 2
      "IOVARendererID" = 16908292
      "IOProviderClass" = "IOPCIDevice"
      "IOCFPlugInTypes" = {"ACCF0000-0000-0000-0000-000a2789904e"="IOAccelerator2D.plugin"}
      "IOGVACodec" = "AMDVCN2"
      "SWIP_properties" = {"EnableSwVCNFWLoading"=0}
      "IOGVAHEVCDecode" = "1"
      "IOClass" = "AMDRadeonX6000_AMDNavi14GraphicsAccelerator"
      "CFBundleIdentifierKernel" = "com.apple.kext.AMDRadeonX6000"
      "IOReportLegendPublic" = Yes
      "GPUConfigurationVariable" = {"NumROP"=8,"MemBusWidth"=0,"NumCUPerSH"=10,"NumSH"=2,"SysClkFreq"=190000,"NumSE"=1,"MemClkFreq"=87500}
      "IOPCITunnelCompatible" = Yes
      "IOPCIMatch" = "0x73401002 0x73411002 0x73431002 0x73471002"
      "IOMatchCategory" = "IOAccelerator"
      "IOReportLegend" = ({"IOReportChannels"=((4000,6442450945,"Overall GPU Busy"),(4001,6442450945,"Overall GPU total time"),(4002,6442450945,"3d Busy"),(4003,6442450945,"3d total time"),(4010,6442450945,"DMA Busy"),(4011,6442450945,"DMA total time"),(4006,6442450945,"Video Busy"),(4007,6442450945,"Video total time"),(4008,6442450945,"Compute Busy"),(4009,6442450945,"Compute total time"),(4705492258585592320,6442450945,"GFX busy"),(4705492258585596928,6442450945,"GFX total time"),(4705492241555999810,6442450945,"Compute low priority busy"),(4705492241555999828,6442450945,"Compute low priority total time"),(4705492241556000066,6442450945,"Compute medium priority busy"),(4705492241556000084,6442450945,"Compute medium priority total time"),(4705492241555998786,6442450945,"Compute high priority busy"),(4705492241555998804,6442450945,"Compute high priority total time"),(4705492245816619074,6442450945,"DMA 0 busy"),(4705492245816619092,6442450945,"DMA 0 total time"),(4705492245816619330,6442450945,"DMA 1 busy"),(4705492245816619348,6442450945,"DMA 1 total time"),(0,6442450945,"Command Buffer Bytes"),(1,6442450945,"GL Command Buffer Bytes"),(2,6442450945,"CL Command Buffer Bytes"),(3,6442450945,"2D Command Buffer Bytes"),(4,6442450945,"Video Command Buffer Bytes"),(5,6442450945,"Out-Of-Line Texture Creation Bytes"),(6,6442450945,"Shared System Memory Texture Creation Bytes"),(7,6442450945,"Private Video Memory Texture Creation Bytes"),(8,6442450945,"Private System Memory Texture Creation Bytes"),(9,6442450945,"IOSurface Backed Texture Creation Bytes"),(10,6442450945,"Out-Of-Line Texture Creation Count"),(11,6442450945,"Private Video Memory Texture Creation Count"),(12,6442450945,"Private System Memory Texture Creation Count"),(13,6442450945,"Shared System Memory Texture Creation Count"),(14,6442450945,"IOAccelSurfaceRef Texture Creation Count"),(15,6442450945,"IOAccelSurfaceBufferRef Texture Creation Count"),(16,6442450945,"IOSurface Backed Texture Creation Count"),(17,6442450945,"Texture Count"),(20,6442450945,"Bytes mapped into the GPU"),(21,6442450945,"Bytes unmapped from the GPU"),(82,6442450945,"IOAccelClientShared allocated bytes"),(22,6442450945,"Private textures bytes paged on"),(23,6442450945,"Out of line texture bytes paged on"),(24,6442450945,"IOAccelSurface bytes paged on"),(25,6442450945,"IOSurface bytes paged on"),(26,6442450945,"IOSurface bytes evicted"),(27,6442450945,"IOAccelSurface bytes evicted"),(28,6442450945,"Texture bytes evicted"),(29,6442450945,"IOSurface bytes paged off"),(30,6442450945,"IOAccelSurface bytes paged off"),(31,6442450945,"Texture bytes paged off"),(32,6442450945,"Texture bytes voluntarily paged off"),(33,6442450945,"IOAccelSurface count"),(34,6442450945,"Volatile IOAccelSurface count"),(35,6442450945,"2D context count"),(36,6442450945,"Video context count"),(37,6442450945,"OpenGL context count"),(38,6442450945,"OpenCL context count"),(39,6442450945,"Buffer swap count"),(40,6442450945,"Buffer flip count"),(41,6442450945,"Databuffer count"),(42,6442450945,"VRAM eviction wait time"),(43,6442450945,"CPU wait on GPU"),(44,6442450945,"CPU wait to submit to GPU"),(45,6442450945,"GL Client wait on GPU"),(46,6442450945,"CL Client wait on GPU"),(47,6442450945,"CPU Wait to evict resource"),(49,6442450945,"CPU wait for free databuffer"),(53,6442450945,"GL Client wait for swap complete"),(55,6442450945,"CPU Wait to copy in surface"),(56,6442450945,"CPU Wait to copy out surface"),(59,6442450945,"CPU Wait for surface idle"),(60,6442450945,"CPU Wait for resource read"),(61,6442450945,"CPU Wait for resource write"),(62,6442450945,"CPU Wait for all 2D finish"),(63,6442450945,"CPU Wait for 2D finish"),(64,6442450945,"CPU Wait for video finish"),(65,6442450945,"CPU Wait for GL finish"),(66,6442450945,"CPU Wait for CL finish"),(106,6442450945,"In use system memory"),(105,6442450945,"In use video memory"),(103,6442450945,"Command Queue Count"),(104,6442450945,"GPU Restart Count")),"IOReportGroupName"="Internal Statistics","IOReportChannelInfo"={"IOReportChannelUnit"=0}})
      "IOProbeScore" = 0
      "IOGVAHEVCEncodeCapabilities" = {"VTPerProfileDetails"={"1"={"VTMaxEncodeLevel"=153}},"VTQualityRating"=50,"VTRating"=350,"VTSupportedProfileArray"=(1)}
      "IOAccelDisplayPipeCapabilities" = {"DisplayPipeSupported"=Yes,"TransactionsSupported"=Yes}
      "SurfaceList" = ()
    }

**Comment by goodspeed** [comment]
hi @exelban, can you reopen this issue? I don't want to post a new issue. 

I replied above.

Thanks.

**Comment by exelban** [comment]
Hi. Thanks for sharing the output.
According to the output, temperature and utilization must be available.
Did the RX 5500 is enabled?

**Comment by goodspeed** [comment]
Yes, the RX 5500 is installed. 

This is on the hackintosh. I have the AMD Power Gadget working with AMDRyzenCPUPowerManagement.kext & SMCAMDProcessor.kext. 

It was working normally with the other hackintosh Radeon RX 580. But when I setup on another machine with the RX 5500XT, the GPU temp, fan speed & util have no values. 

What can I check or enable?

**Comment by exelban** [comment]
Nothing. I will make a build with a log to file. It will allow debugging the problem.

**Comment by goodspeed** [comment]
Thanks so much for your great work.

**Comment by exelban** [comment]
Hi. Could you please run this build for a few minutes?
After you will find a file `log.txt` in the `Documents` folder. Could you please share this file?

[Stats.dmg.zip](https://github.com/exelban/stats/files/6503023/Stats.dmg.zip)

**Comment by goodspeed** [comment]
Here it is.

[log.txt](https://github.com/exelban/stats/files/6511763/log.txt)

**Comment by goodspeed** [comment]
![Screen Shot 2021-05-20 at 6 21 03 AM](https://user-images.githubusercontent.com/11496/118892616-c258c400-b933-11eb-88de-079059a1b120.png)

What is missing are temp, fan speed, core clock, mem clock (not utilization)

**Comment by exelban** [comment]
Ok. Your driver just not provide this information about GPU. Not all drivers provide the right stats. In your case driver have these stats in the API, but all have 0 value. There is no way I can fix it.

**Comment by goodspeed** [comment]
Thanks @exelban
